### ISSUE TITLE
SessionDestroyed event triggered more times

### ISSUE THREAD

**Issue by fuinumir** [spring-projects/spring-session#292]
Hi,
I'm trying the new class SessionEventHttpSessionListenerAdapter, in order to support my HttpSessionListener with SpringSession (1.1.0-BUILD-SNAPSHOT)

I registered my listener in the adapter:
HttpSessionConfig.java
```java  
@Bean
    public SessionEventHttpSessionListenerAdapter sessionEventHttpSessionListenerAdapter() {
        List<HttpSessionListener> demoSessionListeners = new ArrayList<HttpSessionListener>();
        demoSessionListeners.add(new DemoSessionListener());
        return new SessionEventHttpSessionListenerAdapter(demoSessionListeners);
    }
```
I created a sample listener that logs the creation and the destruction of the session
DemoSessionListener.java
```java
public class DemoSessionListener implements HttpSessionListener {

    @Override
    public void sessionCreated(HttpSessionEvent httpSessionEvent) {
        System.out.println("sessionCreated with id " + httpSessionEvent.getSession().getId());
    }

    @Override
    public void sessionDestroyed(HttpSessionEvent httpSessionEvent) {
        System.out.println("sessionDestroyed with id " + httpSessionEvent.getSession().getId());
    }
```

The sessionDestroyed event is triggered more times:
Output:
```
sessionDestroyed with id 9f8eb544-cced-4e93-95be-e2df9b8b5424
sessionDestroyed with id 9f8eb544-cced-4e93-95be-e2df9b8b5424
sessionCreated with id :fa9bf22e-8162-46ee-b355-ec787809e7a0
```

I think the event should be triggered only once time.
Thanks in advance

**Comment by rwinch** [comment]
Any chance you are using `@EnableRedisHttpSession` and exposed `DemoSessionListener` as a Bean?  If so, then you do not need to explicitly add a `SessionEventHttpSessionListenerAdapter` with your `DemoSessionListener` added. This is because `@EnableRedisHttpSession` will automatically register any Bean of type `HttpSessionListener`.

**Comment by grayros** [comment]
Mm,me too. I just exposed my Listener as a Bean,using <code>RedisHttpSessionConfiguration</code>.

**Comment by grayros** [comment]
One session id with two object.
![screenshot_1](https://cloud.githubusercontent.com/assets/15182190/10657023/ff377d6e-78af-11e5-9e3f-dee3bd8c0153.png)
![screenshot_2](https://cloud.githubusercontent.com/assets/15182190/10657027/06629218-78b0-11e5-90cc-44aa5a125729.png)

**Comment by rwinch** [comment]
I'm seeing this behavior as well. Will fix for 1.1.0 M1
